Source/SocketIO/Client/SocketIOClient.swift

```diff
@@ -213,22 +213,44 @@ open class SocketIOClient : NSObject, SocketIOClientSpec {
     /// - parameter items: The items to send with this event. May be left out.
     open func emit(_ event: String, _ items: SocketData...) {
         do {
-            try emit(event, with: items.map({ try $0.socketRepresentation() }))
+            try emit(event, with: items.map({ try $0.socketRepresentation() }), completion: {})
         } catch {
             DefaultSocketLogger.Logger.error("Error creating socketRepresentation for emit: \(event), \(items)",
                                              type: logType)
 
             handleClientEvent(.error, data: [event, items, error])
         }
     }
+    
+    /// Send an event to the server, with optional data items and write completion handler.
+    ///
+    /// If an error occurs trying to transform `items` into their socket representation, a `SocketClientEvent.error`
+    /// will be emitted. The structure of the error data is `[eventName, items, theError]`
+    ///
+    /// - parameter event: The event to send.
+    /// - parameter items: The items to send with this event. May be left out.
+    /// - parameter completion: Callback called on transport write completion.
+    open func emit(_ event: String, _ items: SocketData..., completion: @escaping () -> ())  {
+        emit([event] + items, completion: completion)
```

Comment:
This needs to do the transformation from SocketData to something representable in JSON. See the other swift-centric methods.

Reply:
Ah yea of course, added.